### ISSUE TITLE
Setting of the weight of gradient penalty

### ISSUE THREAD

**Issue by mrluin** [Zeleni9/pytorch-wgan#26]
Hello,

Thanks for your great implementation!

Is there some tricks for setting the weight of gp? I noticed that the original paper adopted weight of 10, but when I used the same weight on my own data(super-resolution task), it was hard for critic to distinguish real/fake data, that means d_real_loss and d_fake_loss are nearly the same. After I lower the weight of gp, the training process seems have a better behavior.

So, the weight of gradient penalty can only be set empirically? Do you have some good experience or tricks for it?

Thanks in advance!

**Comment by Zeleni9** [comment]
Hey, when I written implementation I didn't play with that parameters. But it seems it has to with regularization for wgan-gradient penalty. I can c/p a part from the paper [On the Regularization of Wasserstein GAN](https://arxiv.org/abs/1709.08894).

In particular, for small &#x03BB; the optimal scores approximately agree. On the other hand, increasing
&#x03BB; strengthens the soft constraints, which means that the theoretical observations from Section 4
become more pertinent with growing &#x03BB;. Our experiments show exactly the behavior that WGANLP
and WGAN-GP perform very similarly for small &#x03BB;, while WGAN-LP performs much better for
larger values of &#x03BB; and its performance is much less dependent on the choice of hyperparameter &#x03BB;.
